### ISSUE TITLE
[Crash]: Going to the nether in offline world attempted to connect me to a server and crashed my game.

### ISSUE THREAD

**Issue by TheLeadZombie** [AllTheMods/ATM-10#1447]
### Possible Fixes

Yes

### Modpack Version

2.4

### What happened?

Like why tf is it trying to connect me to server in an offline world? I tried again and it worked fine but the fact it tried at all pisses me off.

### Crash Log

https://gist.github.com/TheLeadZombie/ef78e4c061fd3c1918620e57d95b282b

**Comment by alekthefirst** [comment]
When playing on a local world you're playing on an integrated server, that works much the same as any other server. Kicking you out to the multiplayer screen is something that can happen of it crashes, and if it turns out to be reproducible it is certainly of interest. Looks to have been caused by an interaction between Sodium and Iris regarding cancelling cloud rendering.

**Comment by DaddyLongspoon** [comment]
yep the server is a seperate process in another thread running on the same machine as the client (the application you're interacting with). so this is an asychronisation between the two threads. happens occasionally even on very stable packs.  Try allocating more ram or shutting down other applications (do you have a browser window with forty odd tabs open ;)

**Comment by DaddyLongspoon** [comment]
> > yep the server is a seperate process in another thread running on the same machine as the client (the application you're interacting with). so this is an asychronisation between the two threads. happens occasionally even on very stable packs. Try allocating more ram or shutting down other applications (do you have a browser window with forty odd tabs open ;)
> 
> i wouldn't think it's a ram issue since i have allocated ~30gb out of the 64gb i have total. but it would probably help if i stopped tab hoarding lmao. It does make me curious, was minecraft always this way or is like the local server thing new? I mean, i guess when i think about it, making it local server/host probably does make things easier.

in the earliest builds it did not work this way but the multiplayer game had to be a seperate build, different application.  this was in beta days, I purchased minecraft very early for two dollars or so.    but it didn't take long at all for notch to adopt this model. don't recall exactly which version but maybe only a month or two after I first got it

**Comment by TheLeadZombie** [comment]
yeah i think they had separate launcher for servers i think at one point
